pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -259,6 +259,25 @@ stdenv.mkDerivation {
     inherit bintools;
     inherit cc libc libcxx nativeTools nativeLibc nativePrefix isGNU isClang;
 
+    # Expose the C++ standard library we're using. See the comments on "General
+    # libc++ support". This is also relevant when using older gcc than the
+    # stdenv's, as may be required e.g. by CUDAToolkit's nvcc.
+    cxxStdlib =
+      let
+        givenLibcxx = libcxx.isLLVM or false;
+        givenGccForLibs = useGccForLibs && gccForLibs.langCC or false;
+      in
+      if (!givenLibcxx) && givenGccForLibs then
+        { kind = "libstdc++"; lib = gccForLibs; }
```

Comment:
Nit: should this be `gccForLibs.lib` to be of the same "kind" as `libcxx`? This can't be `gccForLibs_solib` because the latter is a string (e.g. can't `gccForLibs_solib.version`)